(trunk) KAFKA-5232 Fix Log.parseTopicPartitionName to take into account dot character in topic names of deleted topics

jaikiran opened the pull request:
The commit here contains a fix and a test case for the issue reported in https://issues.apache.org/jira/browse/KAFKA-5232. This PR is meant for `trunk` branch and a corresponding PR for `0.10.2` branch has been raised here https://github.com/apache/kafka/pull/3043

jaikiran committed:
KAFKA-5232 Fix the bug where the Log.parseTopicPartitionName wasn't taking into account the potential presence of '.' character in topic names while looking for topics that are marked for deletion

ijuma commented on core/src/test/scala/unit/kafka/log/LogTest.scala:
```diff
@@ -1640,6 +1640,18 @@ class LogTest {
     assertEquals(partition.toInt, topicPartition.partition)
   }
 
+  // see https://issues.apache.org/jira/browse/KAFKA-5232
```
This is simple enough that we can write a short description here instead of referencing the JIRA.

ijuma commented on core/src/test/scala/unit/kafka/log/LogTest.scala:
```diff
@@ -1640,6 +1640,18 @@ class LogTest {
 
+  // see https://issues.apache.org/jira/browse/KAFKA-5232
+  @Test
+  def testParseTopicPartitionNameForDeletedTopic() {
```
`testParseTopicPartitionNameWithPeriodForDeletedTopic`?

ijuma commented on core/src/test/scala/unit/kafka/log/LogTest.scala:
```diff
@@ -1640,6 +1640,18 @@ class LogTest {
+    val topicPartition = Log.parseTopicPartitionName(dir)
+    assertEquals("Unexpected topic name parsed", topic, topicPartition.topic)
+    assertEquals("Unexpected partition number parsed", partition.toInt, topicPartition.partition)
+  }
```
While we're doing this, it would be better to flesh out tests for the deleted case. Namely variants of `testParseTopicPartitionNameForMissingSeparator`, `testParseTopicPartitionNameForMissingTopic` and `testParseTopicPartitionNameForMissingPartition` with the `-delete` suffix.

ijuma reviewed (commented):
Thanks for the PR. Left a few suggestions when it comes to the tests.

jaikiran commented:
@ijuma, thanks for reviewing. I have updated this PR to take into account your review comments. I updated those existing test methods to include checks for the `-delete` variant, instead of creating separate test methods for each of those existing ones. However, I am open to create new test methods for them, if you think that's a better idea.

ijuma committed:
Various improvements to `parseTopicPartitionName` and tests

ijuma commented:
@jaikiran, I did a PR to your branch with additional improvements: https://github.com/jaikiran/kafka/pull/1

Please merge if they look good to you.

jaikiran committed:
Merge pull request #1 from ijuma/KAFKA-5232

Various improvements to `parseTopicPartitionName` and tests

jaikiran commented:
Done. Those changes looked good to me.
